### DISCUSSION TITLE
Bug - Can't set value of the 'flex' input on the 'ColumnComponent' component

### DISCUSSION THREAD

**Discussion by pbaradziej** [gund/ng-dynamic-component#500] [Q&A]
Hello,
After updating to version 10.3.0, whenever I open component with ndcDynamic, console error shows up. 
 
Can't set value of the 'flex' input on the 'ColumnComponent' component. Make sure that the 'flex' property is annotated with @Input() or a mapped @Input('flex') exists.

I figured out that it shows because I don't have @Input flex in ColumnComponent, because I don't need it there.
Only RowComponent uses @Input flex.

 <ndc-dynamic [ndcDynamicComponent]="getClassByType()"
               [ndcDynamicInputs]="{
                  components: components,
                  justifyContent: justifyContent,
                  flex: flex,
                  title: title
               }">
 </ndc-dynamic>

After checking your changes i figured out it is because of compRef.setInput line.
![Zrzut ekranu 2023-02-14 154718](https://user-images.githubusercontent.com/44201479/218772008-da0bdf44-0855-405a-bdbe-8bac5cfb47af.png)

Do you plan on potentially fixing the problem of passing Inputs that are not used by every dynamic component, or is it my configuration error?

**Comment by gund** [comment]
It's an error on your side as you are passing inputs which has `flex` property but the component does not have this as input so Angular throws an error.
Previously such cases were silently ignored as this library was setting inputs manually so it did not check if the input actually existed but now Angular introduced new API to set inputs which this library is using and so the error is thrown if there is no input on component.
Just remove the `flex` prop from inputs or add it to the component and you will be fine.
